cpp/ql/src/semmle/code/cpp/models/implementations/Pure.qll

```diff
@@ -54,4 +54,51 @@ class PureFunction extends ArrayFunction, TaintFunction {
       output.isOutReturnValue()
     )
   }
+
+  override predicate parameterNeverEscapes(int i) {
+    getParameter(i).getUnspecifiedType() instanceof PointerType
+  }
+
+  override predicate parameterEscapesOnlyViaReturn(int i) {
+    none()
```

Comment:
The [strstr](https://linux.die.net/man/3/strstr) family of functions return a pointer that's derived from (and may be equal to) their first parameter.